Core/Security/VOMSService.py

```diff
@@ -1,46 +1,72 @@
-# $HeadURL$
+""" VOMSService class encapsulated connection to the VOMS service for a given VO
+"""
+
 __RCSID__ = "$Id$"
 
 from DIRAC import gConfig, S_OK, S_ERROR
 from DIRAC.Core.Utilities.SOAPFactory import getSOAPClient
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getVOOption
 
 def _processListReturn( soapReturn ):
   data = []
-  for entry in soapReturn:
-    data.append( str( entry ) )
+  if soapReturn:
+    for entry in soapReturn:
+      data.append( str( entry ) )
   return data
 
 def _processListDictReturn( soapReturn ):
   data = []
-  for entry in soapReturn:
-    entryData = {}
-    for info in entry:
-      entryData[ info[0] ] = str( info[1] )
-    data.append( entryData )
+  if soapReturn:
+    for entry in soapReturn:
+      entryData = {}
+      for info in entry:
+
+        try:
+          entryData[ info[0] ] = str( info[1] )
+        except:
+          pass
+      data.append( entryData )
   return data
 
 class VOMSService:
 
-  def __init__( self, adminUrl = False, attributesUrl = False, certificatesUrl = False ):
+  def __init__( self, vo, adminUrl = False, attributesUrl = False, certificatesUrl = False ):
```

Comment:
  True. However, the VO is now a
      crucial attribute of objects of this class. I think, I will leave
      only the vo argument as a mandatory one and will make the
      discovery of connection details
      inside this class.
      
      On 28/10/15 11:24, fstagni wrote:
    
    
      In Core/Security/VOMSService.py:
      >    return data
>  
>  class VOMSService:
>  
> -  def __init__( self, adminUrl = False, attributesUrl = False, certificatesUrl = False ):
> +  def __init__( self, vo, adminUrl = False, attributesUrl = False, certificatesUrl = False ):

      This change in signature is a bit dangerous. The class is used
        also in extensions.
      —
        Reply to this email directly or view
          it on GitHub.